### PR TITLE
Migrate EventCard to Shadcn

### PR DESCRIPTION
**Pull request by minimalsm**
## Description

- Updates EventCard to shadcn

## Notes

1. Hardcoded 1.6rem line height for the description text (per OldText) as the defaults we have varied too much from the current design. Our designers should pull this component in line with current DS.

2. Slight differences in spacing compared to the original design (which had kind of unusual spacing). IMO, these are improvements but defer to @nloureiro

3. I've hardcoded the background colors (`bg-[#FCFCFC] dark:bg-[#272627]`) in the CardHeader. Our default theme values didn't provide a good match for this use case. another area where we might want to consult with design

## Related Issue

#13946

### CODE REVIEW

**Comment by nloureiro** [comment]
agree. The spacing was off. Thank you for the adjustments 

comapring with Figma, the text size is bigger, but now we are bumping everything to 16 or above so let's keep it like this

Screen Shot 2024-09-25 11 17 39 PM

With this text size, we should do a 2-card line. It's too narrow for an MD page. Would it be on another PR to do this?

Screen Shot 2024-09-25 11 22 51 PM

**Comment by minimalsm** [comment]
Thanks @nloureiro 

> 
> 
> agree. The spacing was off. Thank you for the adjustments 
> 
> 
> 
> comapring with Figma, the text size is bigger, but now we are bumping everything to 16 or above so let's keep it like this
> 

Okay can update to 16px fontsize. With that we can probably normalise line height too?

> 
> With this text size, we should do a 2-card line. It's too narrow for an MD page. Would it be on another PR to do this?
> 
> 

We can handle that in #13967, which is the upcoming events grid wrapper

**Comment by nloureiro** [comment]
> Thanks @nloureiro
> 
> > agree. The spacing was off. Thank you for the adjustments
> > comapring with Figma, the text size is bigger, but now we are bumping everything to 16 or above so let's keep it like this
> 
> Okay can update to 16px fontsize. With that we can probably normalise line height too?

yes, makes sense
> 
> > With this text size, we should do a 2-card line. It's too narrow for an MD page. Would it be on another PR to do this?
> 
> We can handle that in #13967, which is the upcoming events grid wrapper

ok, got it

**Comment by minimalsm** [comment]
Updated font size implementation after discussion with @nloureiro:

Mobile: 16px font size
Desktop: Retaining 14px font size

Rationale:

- 3-column grid on large screens necessitates smaller font due to limited width
- 2-column grid on large screens was tested but felt clunky and reduced event discoverability

Changes implemented && ready for review :)

File: src/components/EventCard.tsx [hunk 1]
```diff
@@ -28,97 +21,52 @@ const EventCard = ({
...
-    <Box
-      className={className}
-      position="relative"
-      mt="0"
-      _before={clearStyles}
-      _after={clearStyles}
-      w="full"
-      h="full"
+    <Card
+      className={`border-lightBorder flex h-full flex-col rounded-md border ${className}`}
```
Comment (pettinarip) [inline src/components/EventCard.tsx:38]: We should use the `cn` utility to merge classes correctly.
```suggestion
      className={cn("border-lightBorder flex h-full flex-col rounded-md border", className)}
```

**Review by pettinarip — APPROVED** [review]
@minimalsm looks good, will add a small fix to use `cn` function.

File: src/components/EventCard.tsx (hunk 1 above)
Comment (pettinarip) [inline src/components/EventCard.tsx:38]: Ah, what did you mean with `lightBorder`?

Comment (minimalsm) [reply src/components/EventCard.tsx]: No idea, just going to remove lol

File: src/components/EventCard.tsx [hunk 2]
```diff
@@ -28,97 +21,52 @@ const EventCard = ({
...
-    <Box
-      className={className}
-      position="relative"
-      mt="0"
-      _before={clearStyles}
-      _after={clearStyles}
-      w="full"
-      h="full"
+    <Card
+      className={cn("border-lightBorder flex h-full flex-col rounded-md border", className)}
```
Comment (minimalsm) [inline src/components/EventCard.tsx:38]:
```suggestion
      className={cn("flex h-full flex-col rounded-md border", className)}
```

File: src/components/EventCard.tsx [hunk 3]
```diff
@@ -2,24 +2,17 @@ import React from "react"
 import { useRouter } from "next/router"
 import { useTranslation } from "next-i18next"
 import { BsCalendar3 } from "react-icons/bs"
-import { Box, Flex, Heading, Icon } from "@chakra-ui/react"
 
 import type { EventCardProps } from "@/lib/types"
```
Comment (minimalsm) [inline src/components/EventCard.tsx:6]:
```suggestion
import { cn } from "@/lib/utils/cn"

import type { EventCardProps } from "@/lib/types"
```

**Review by pettinarip — APPROVED** [review]

**Pull request merged**